src/beliefs/discrete.jl

```diff
@@ -1,133 +1,107 @@
-# Maintained by Max Egorov and Zach Sunberg
+# Goals: minimize calls to ordered_states (allocates memory)
 
-"""
-A weight vector to be used as a belief or state distribution.
-
-sum(b)=1 is not enforced at all times, but during the update it is normalized to sum to one.
-
-pdf(b, i) calculates the sum every time it is called. To access the weight directly (for example if you are sure that the sum is 1), use weight(b, i). 
-"""
-mutable struct DiscreteBelief
+# needs pomdp for state_index in pdf(b, s)
+# needs list of ordered_states for rand(b)
+mutable struct DiscreteBelief{P<:POMDP, S}
+    pomdp::P
+    state_list::Vector{S}       # vector of ordered states
     b::Vector{Float64}
 end
+function DiscreteBelief(pomdp)
```

Comment:
I don't think we should have this constructor like it is. If I called this, I would not necessarily expect it to return a uniform belief; in fact I think it would be more reasonable to return something consistent with `initial_state_distribution(pomdp)`.

Instead of this, I think we should just have a function called `uniform_belief(pomdp)` or `uniform_discrete_belief(pomdp)` that does the same thing.

It might also be reasonable to have a `DiscreteBelief(pomdp::POMDP, b::Vector{Float64})` constructor that automatically creates the state_list for you.

Reply:
I'm ok with both `uniform_belief(pomdp)` or something similar. It should be very easy to construct a uniform belief, which is why I originally left `DiscreteBelief(pomdp)` in there. 

I'm also ok with another function that creates the state list for you.